tutorials/elixir-phoenix-on-google-app-engine.md

```diff
@@ -0,0 +1,183 @@
+
+## Create a new app and run it locally
+
+In this section, you will create a new Phoenix app and make sure it runs. If
+you already have an app to deploy, you may use it instead.
+
+1.  Run the `phx.new` task to create a new Phoenix project called
+    "appengine_example". This tutorial will omit Ecto for now.
+
+        mix phx.new appengine_example --no-ecto
+
+    Answer "Yes" when the tool asks you if you want to fetch and install
+    dependencies.
+
+2.  Go into the directory with the new application.
+
+        cd appengine_example
+
+3.  Run the app with the following command:
+
+        mix phx.server
+
+    This compiles your server and runs it on port 4000.
+
+4.  Visit [http://localhost:4000](http://localhost:4000) to see the Phoenix
+    welcome screen running locally on your workstation.
+
+## Deploy your application
+
+Now you will deploy your new app to App Engine.
+
+1.  Create a file called `app.yaml` at the root of the application directory,
+    with the following contents:
+
+        env: flex
+        runtime: gs://elixir-runtime/elixir.yaml
+
+    This configuration selects the Elixir Runtime, an open source App Engine
+    runtime that knows how to build Elixir and Phoenix applications. You can
+    find more information about this runtime at its
+    [Github page](https://github.com/GoogleCloudPlatform/elixir-runtime).
+
+2.  Run the following command to deploy your app:
+
+        gcloud app deploy
+
+    If this is the first time you have deployed to App Engine in this project,
+    gcloud will prompt you for a region. You may choose the default of
+    "us-central", or select a region close to your location.
+
+    The Elixir Runtime will take care of building your application in the
+    cloud, including installing mix dependencies, compiling to BEAM files, and
+    even using brunch to build and digest your assets.
+
+    Deployment will also take a few minutes to requisition and configure the
+    needed resources, especially the first time you deploy.
+
+3.  Once the deploy command has completed, you can run
+
+        gcloud app browse
+
+    to see your app running in production on App Engine.
+
+## Update your application
+
+Let's make a simple change and redeploy.
+
+1.  Open the front page template
+    `lib/appengine_example_web/templates/page/index.html.eex` in your editor.
+    Make a change to the HTML template.
+
+2.  Run the deployment command again:
+
+        gcloud app deploy
+
+    App Engine and the Elixir Runtime will take care of rebuilding your app,
+    deploying the updated version, and switching traffic from the old to the
+    new.
+
+3.  View your changes live by running
+
+        gcloud app browse
+
+## Clean up
+
+After you've finished this tutorial, you can clean up the resources you created
+on Google Cloud Platform so you won't be billed for them in the future. You can
+accomplish this either by deleting the project, or by stopping the App Engine
```

Comment:
Better phrasing: "To clean up the resources, you can delete the project or stop the App Engine service."

Reply:
Done.